openedx/core/djangoapps/user_api/views.py

```diff
@@ -345,10 +350,63 @@ def post(self, request):
             }
             return JsonResponse(errors, status=400)
 
+        password = data.get("password")
+        password_copy = data.get("password_copy")
+        if password and password_copy and password != password_copy:
+            errors = {
+                'password_copy': [{"user_message": "Passwords don't match."}]
+            }
+            return JsonResponse(errors, status=400)
+
         response = JsonResponse({"success": True})
         set_logged_in_cookies(request, response, user)
+
+        if getattr(settings, 'REGISTRATION_EMAIL_FULL_VERIFICATION', None) is not None:
```

Comment:
After some conversation with my fellow engineer in edX, we believe the best way for you to incorporate your email address validation is to develop a hook on this platform, so that you can invoke a different module on your server environment to actual perform the validation.
One example (strictly an example) is to develop a micro service app, that does all the email address validation checks you written here, and expose api (RESTful, RPC, RabbitMQ) for edx-platform to send validation requests.